lib/plugin.go

```diff
@@ -0,0 +1,34 @@
+package lib
+
+import (
+	"fmt"
+	goplugin "plugin"
+
+	"github.com/loadimpact/k6/plugin"
+)
+
+func LoadJavaScriptPlugin(path string) (plugin.JavaScriptPlugin, error) {
+	p, err := loadPlugin(path)
+	if err != nil {
+		return nil, err
+	}
+
+	jsSym, err := p.Lookup("JavaScriptPlugin")
+	if err != nil {
+		return nil, err
+	}
+
+	var jsPlugin plugin.JavaScriptPlugin
+	jsPlugin, ok := jsSym.(plugin.JavaScriptPlugin)
+	if !ok {
+		return nil, fmt.Errorf("could not cast plugin to the right type")
+	}
+
+	return jsPlugin, nil
+}
+
+func loadPlugin(path string) (*goplugin.Plugin, error) {
```

Comment:
I want to expand on this (maybe not necessarily for v1) to ensure that we return nice error messages for things like:

- plugin built with a different version of the `plugin` package
- plugin built with a different version of go
- file not found
- wrong architecture